src/commands/createStaticWebApp/StaticWebAppNameStep.ts

```diff
@@ -62,14 +66,11 @@ export class StaticWebAppNameStep extends AzureNameStep<IStaticWebAppWizardConte
     }
 
     private async isSwaNameAvailable(context: IStaticWebAppWizardContext, resourceName: string): Promise<boolean> {
-        // Static Web app names must be unique to the current resource group.
-        try {
-            await context.client.staticSites.getStaticSite(resourceName, resourceName);
-            return false;
+        // Our implementation is that Static Web app names must be unique to the subscription.
+        listOfSites = listOfSites || await uiUtils.listAll(context.client.staticSites, context.client.staticSites.list());
```

Comment:
I'm having a hard time deciding on this PR. I'm just not sure listing all static sites is a robust or efficient way to solve the root cause. We've discussed a few alternative solutions - what were those and why is this one better?

Reply:
The main alternative was to put the resource group step first so we can be certain whether or not the swa is within that rg.

Since we don't display swa with duplicate names very well, enforcing a unique name helps mitigate that problem.
![image](https://user-images.githubusercontent.com/5290572/126560846-a9d2e04a-df0f-4226-b0ec-1e4b10ecec45.png)

Reply:
> Since we don't display swa with duplicate names very well

Just to be clear, we're talking about "ReactSite", right?  Seems like the description already differentiates them pretty well. I believe the "Invalid" one is an unrelated issue. 

> The main alternative was to put the resource group step first so we can be certain whether or not the swa is within that rg.

Regardless of the final solution, I think we should make sure it only affects advanced create. Basic create is fine as-is since we intelligently pick the rg for them anyways. Now, in terms of advanced create, we could:
1. Enforce swa name uniqueness at the subscription level (aka your current changes, except only for advanced create)
1. Move the rg step
1. Filter out rgs that would have a conflicting name
1. Validate the name/rg immediately after they pick an rg and show a warning/error at that point
1. Don't even try to validate uniqueness and just go with whatever error is thrown by the platorm

I think I lean towards 2 or 4. I'm just trying to balance code complexity and perf impacts for something that is really just an edge case bug. Also, it would be nice if we're consistent with the portal in terms of name uniqueness rules

Reply:
I went with option 2.  Just felt the best to me.